### ISSUE TITLE
unused-import bug (false-positive)

### ISSUE THREAD

**Issue by elseecay** [PyCQA/pylint#8177]
### Bug description

```python
# main.py

from pack import subpack
import pack.subpack.subpackmodule

subpack.subpackmodule.f()
```

### Command used

```shell
pylint --disable=all --enable=unused-import ./main.py
```

### Pylint output

```shell
************* Module main
main.py:2:0: W0611: Unused import pack.subpack.subpackmodule (unused-import)

-------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 10.00/10, -3.33)
```

### Expected behavior

```shell
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Pylint version

```shell
pylint 2.16.0
astroid 2.14.1
Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]
```

### OS / Environment

Ubuntu 22.04

**Comment by elseecay** [comment]
Real case was:

```python
from app.storage import sql
# pylint: disable-next=unused-import
import app.storage.sql.content
# pylint: disable-next=unused-import
import app.storage.sql.manifest
# pylint: disable-next=unused-import
import app.storage.sql.description
# pylint: disable-next=unused-import
import app.storage.sql.share
# pylint: disable-next=unused-import
import app.storage.sql.impexp
# pylint: disable-next=unused-import
import app.storage.sql.raw
```

I dont wanna all this names like content, manifest etc. appears in the global scope with
`from app.storage.sql import content`
instead access to it with
`sql.content`

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening an issue. I think it's a duplicate of one of the oldest issue in the repo: https://github.com/PyCQA/pylint/issues/241
